src/vs/workbench/contrib/webview/browser/pre/main.js

```diff
@@ -204,7 +204,7 @@ const workerReady = new Promise((resolve, reject) => {
 		return reject(new Error('Service Workers are not enabled. Webviews will not work. Try disabling private/incognito mode.'));
 	}
 
-	const swPath = `service-worker.js${self.location.search}`;
+	const swPath = `service-worker.js?vscode-resource-base-authority=${(new URL(location.toString()).searchParams).get('vscode-resource-base-authority')}`;
```

Comment:
[nit] We already have a url object on this line: https://github.com/microsoft/vscode/blob/bbc10afedac1f9ea35d510211a5ec59296573cd5/src/vs/workbench/contrib/webview/browser/pre/main.js#L21

Let's reusing it